test/Parse/ConditionalCompilation/basicParseErrors.swift

```diff
@@ -139,6 +139,24 @@ func undefinedFunc() // ignored.
 #endif
 undefinedFunc() // expected-error {{use of unresolved identifier 'undefinedFunc'}}
 
+#if os(simulator) // expected-warning {{unknown operating system for build configuration 'os'}} expected-note {{did you mean 'targetEnvironment'}} {{5-7=targetEnvironment}}
+#endif
+
+#if arch(iOS) // expected-warning {{unknown architecture for build configuration 'arch'}} expected-note {{did you mean 'os'}} {{5-9=os}}
+#endif
+
+#if _endian(arm64) // expected-warning {{unknown endianness for build configuration '_endian'}} expected-note {{did you mean 'arch'}} {{5-12=arch}}
+#endif
+
+#if targetEnvironment(_ObjC) // expected-warning {{unknown target environment for build configuration 'targetEnvironment'}} expected-note {{did you mean '_runtime'}} {{5-22=_runtime}}
```

Comment:
Come to think of it, we probably *shouldn't* suggest underscored build configurations—they're not officially supported, after all.

Reply:
Makes sense. This is addressed in commit [e44a458](https://github.com/apple/swift/pull/25977/commits/e44a458ec8df83275426b1db4c828439b53325c1).